collect_app/src/main/java/org/odk/collect/android/widgets/utilities/DateTimeWidgetUtils.java

```diff
@@ -0,0 +1,104 @@
+package org.odk.collect.android.widgets.utilities;
+
+import android.os.Build;
+import android.os.Bundle;
+
+import org.javarosa.core.model.FormIndex;
+import org.javarosa.core.model.data.TimeData;
+import org.joda.time.DateTime;
+import org.joda.time.LocalDateTime;
+import org.odk.collect.android.activities.FormEntryActivity;
+import org.odk.collect.android.fragments.dialogs.BikramSambatDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.CopticDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.CustomDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.CustomTimePickerDialog;
+import org.odk.collect.android.fragments.dialogs.EthiopianDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.FixedDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.IslamicDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.MyanmarDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.PersianDatePickerDialog;
+import org.odk.collect.android.logic.DatePickerDetails;
+import org.odk.collect.android.utilities.DialogUtils;
+import org.odk.collect.android.utilities.ThemeUtils;
+
+public class DateTimeWidgetUtils {
+
+    private DateTimeWidgetUtils() {
+    }
+
+    public static TimeData getTimeData(int hourOfDay, int minuteOfHour) {
+        // use picker time, convert to today's date, store as utc
+        DateTime localDateTime = new DateTime()
+                .withTime(hourOfDay, minuteOfHour, 0, 0);
+
+        return new TimeData(localDateTime.toDate());
+    }
+
+    public static LocalDateTime getCurrentDate() {
+        return LocalDateTime
+                .now()
+                .withHourOfDay(0)
+                .withMinuteOfHour(0)
+                .withSecondOfMinute(0)
+                .withMillisOfSecond(0);
+    }
+
+    public static void showDatePickerDialog(FormEntryActivity activity, FormIndex formIndex, DatePickerDetails datePickerDetails,
+                                            LocalDateTime date) {
+        ThemeUtils themeUtils = new ThemeUtils(activity);
+
+        Bundle bundle = new Bundle();
+        bundle.putInt(CustomDatePickerDialog.DATE_PICKER_THEME, getTheme(themeUtils, datePickerDetails));
```

Comment:
Those fields are now hared between CustomDatePickerDialog and FixedDatePickerDialog so shouldn't placed in CustomDatePickerDialog. This class would be probably a better place.